src/rust/ide/lib/analytics/src/remote_log.rs

```diff
@@ -9,27 +9,84 @@ pub use wasm_bindgen::prelude::*;
 
 
 #[wasm_bindgen(inline_js="
-export function _remote_log(msg, data) {
-    if (window !== undefined && window.enso !== undefined && window.enso.remote_log !== undefined) {
-        window.enso.remote_log(msg,{data:data})
+export function _remote_log(msg, value) {
+    if (window !== undefined && window.enso !== undefined && window.enso.remoteLog !== undefined) {
+        try {
+            if (value === undefined) {
+                window.enso.remoteLog(msg)
+            } else {
+                window.enso.remoteLog(msg,{value})
+            }
+
+        } catch (error) {
+            console.error(\"Error while logging message. \" + error );
+        }
+
+    } else {
+        console.warn(\"Failed to send log message.\")
+    }
+}
+
+export function _register(data) {
+    if (window !== undefined && window.enso !== undefined && window.enso.remoteLog !== undefined) {
+        try {
+            window.enso.register(msg,data)
+        } catch (error) {
+            console.error(\"Error while logging message. \" + error );
+        }
+
+    } else {
+        console.warn(\"Failed to send log message.\")
+    }
+}
+
+export function _remote_log_data_field(msg, field_name, value) {
+    if (window !== undefined && window.enso !== undefined && window.enso.remoteLog !== undefined) {
+        try {
+            const data = {}
+            data[field_name] = value
+            window.enso.remoteLog(msg,data)
+        } catch (error) {
+            console.error(\"Error while logging message. \" + error );
+        }
+
     } else {
         console.warn(\"Failed to send log message.\")
     }
 }
 ")]
```

Comment:
1. This became a bit of js. I'd prefer to put it in a js file, not inline snippet
2. I don't know js very well, but is there _any_ chance to reduce all repeated ifs an tries here? For example, to use a helper wrapping some lambda in those if and try?

Reply:
This has been greatly reduced. Especially since one of the methods was unused. 
So, I'd opt for keeping it as a snippet in its new form.